docs/filters.rst

```diff
@@ -121,3 +130,104 @@ logs.  It exposes the following additional methods.
+
+Starting with web3 version 4, the ``watch`` method was taken out of the web3 filter objects.
+There are many decisions to be made when designing a system regarding threading and concurrency.
+Rather than force a decision, web3 leaves these choices up to the user. Below are some example implementations of asynchronous filter event handling that can serve as starting points.
+
+Single threaded concurrency with ``async`` and ``await``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    Beginning in python 3.5, the ``async`` and ``await`` built-in keywords were added.  These provide a shared api for coroutines that can be utilized by modules such as the built-in asyncio_.  Below is an example event loop using asyncio_, that polls a web3 filter object, and passes new entries to a handler.
+
+        .. code-block:: python
+
+            from web3.auto import w3
+            import asyncio
+
+
+            def handle_event(event):
+                print(event)
+                # and whatever
+
+            async def log_loop(event_filter, poll_interval):
+                while True:
+                    for event in event_filter.get_new_entries():
+                        handle_event(event)
+                    await asyncio.sleep(poll_interval)
+
+            def main():
+                block_filter = w3.eth.filter('latest')
+                loop = asyncio.get_event_loop()
+                try:
+                    task = loop.create_task(log_loop(block_filter, 2))
+                    loop.run_until_complete(task)
+                finally:
+                    loop.close()
+
+            if __name__ == '__main__':
+                main()
+
+    Read the asyncio_ documentation for more information.
+
+Running the event loop in a separate thread
+"""""""""""""""""""""""""""""""""""""""""""
+
+    Here is an extended version of above example, where the event loop is run in a separate thread, releasing the ``main`` function for other tasks.
+
+        .. code-block:: python
+
+            from web3.auto import w3
+            import asyncio
+            from threading import Thread
+
+
+            def handle_event(event):
+                print(event)
+                # and whatever
+
+
+            async def log_loop(event_filter, poll_interval):
+                while True:
+                    for event in event_filter.get_new_entries():
+                        handle_event(event)
+                    await asyncio.sleep(poll_interval)
+
+
+            async def start_log_loop(loop):
+                block_filter = w3.eth.filter('latest')
+                asyncio.set_event_loop(loop)
+                task = loop.create_task(log_loop(block_filter, 5))
+                try:
+                    loop.run_until_complete(task)
+                finally:
+                    loop.close()
+
+
+            def main():
+                loop = asyncio.new_event_loop()
+                worker = Thread(target=start_log_loop, args=(loop,))
```

Comment:
We don't have to use asyncio in this other approach, I think it would be clearer if we didn't. For example:

`Thread(target=log_loop, args=(block_filter, 5), daemon=True)`

(where `log_loop` is no longer async, and just uses `time.sleep`)